### PR TITLE
Favorite Charts

### LINKED ISSUE #534: Allow users to favorite charts
### User stories

As a user
When I view the charts table
Then I see a row with a star button
And clicking the star button toggles my "favorited" status with that chart

As a user
When I view the charts table
Then I see my favorited charts first

Image

### PR DESCRIPTION
**Pull request by Elkrival**
closes #534 

This pr adds the feature to favorite a chart. 
We store a list of chart id's to the favoriteCharts property in the user document, the property is used to sort the charts by placing the user's favorites at the top. 
The charts query has been updated from find to an aggregate.
The tests have been updated to support a global mongodb instance.

Screenshot 2023-12-26 at 8 28 27 AM

### CODE REVIEW

File: server/controllers/chartsController/chartsController.test.js [hunk 1]
```diff
@@ -47,59 +51,64 @@ describe('chartsController', () => {
...
+      })
+      afterEach(async () => {
+        await dataDb.collection('charts').drop()
+      })
+
+      it('returns a list of charts in order of user favorite, title, alongside a status of 200', async () => {
+        const [chart1, chart2, chart3] = [
+          createChart({
+            assessment: 'EEG Quick',
+            _id: ObjectId(),
```
Comment (JackVCurtis) [inline server/controllers/chartsController/chartsController.test.js]: Do we need the ID or will it be set automatically?

Comment (JackVCurtis) [reply server/controllers/chartsController/chartsController.test.js]: If we need it maybe build it into the fixture

File: server/models/ChartsModel/index.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 
+const id = '$_id'
+const idKey = '$chart_id'
+const chartId = 'chart_id'
 const ChartsModel = {
   create: async (db, chartAttributes) =>
     await db.collection(collections.charts).insertOne(chartAttributes),
-  index: async (db, query) =>
-    await db.collection(collections.charts).find(query),
+  index: async (db, user) =>
```
Comment (JackVCurtis) [inline server/models/ChartsModel/index.js]: On the user model we use `all` instead of `index`, can we use it here as well for consistency?

**Review by JackVCurtis — APPROVED** [review]
Couple quick things but looks great overall! I'll approve now since I don't think the few fixes will need re-review

File: server/models/ChartsModel/index.js (hunk 2 above)
Comment (Elkrival) [reply server/models/ChartsModel/index.js]: [e39a67e](https://github.com/AMP-SCZ/dpdash/pull/557/commits/e39a67eb3ef48faac109401996dd3d299b3fdc6b)

File: server/controllers/chartsController/chartsController.test.js (hunk 1 above)
Comment (Elkrival) [reply server/controllers/chartsController/chartsController.test.js]: [e39a67e](https://github.com/AMP-SCZ/dpdash/pull/557/commits/e39a67eb3ef48faac109401996dd3d299b3fdc6b)

**Pull request merged**